Formula/m1ddc.rb

```diff
@@ -0,0 +1,37 @@
+class M1ddc < Formula
+  desc "Control external displays (USB-C/DisplayPort Alt Mode) using DDC/CI on M1 Macs"
+  homepage "https://github.com/waydabber/m1ddc"
+  url "https://github.com/waydabber/m1ddc/archive/refs/tags/v1.0.0.tar.gz"
+  sha256 "884b65910e69915db29182363590d663a1a6d983e13ca5c41a74209058084c44"
+  license "MIT"
+
+  head "https://github.com/waydabber/m1ddc.git", branch: "main"
+
+  depends_on arch: :arm
+  depends_on macos: :monterey
+  depends_on :macos
+
+  def install
+    system "make"
+    bin.install "m1ddc"
+  end
+
+  def caveats
+    on_macos do
+      on_intel do
+        <<~EOS
+          m1ddc requires Apple Silicon, consider using ddcctl on Intel instead.
+        EOS
+      end
+    end
+  end
+
+  test do
+    # Ensure helptext is rendered
+    assert_includes shell_output("#{bin}/m1ddc help", 1), "Controls volume, luminance"
+
+    # Attempt getting maximum luminance (usually 100),
+    # will return code 1 if a screen can't be found (e.g., in CI)
+    assert_match(/(\d*)|(Could not find a suitable external display\.)/, pipe_output("#{bin}/m1ddc get luminance"))
```

Comment:
```suggestion
    assert_match(/(\d*)|(Could not find a suitable external display\.)/, shell_output("#{bin}/m1ddc get luminance"))
```

Reply:
[shell_output](https://rubydoc.brew.sh/Homebrew/Assertions#shell_output-instance_method) can't be used here since `m1ddc get luminance` returns status code 1 when a suitable external display can't be found ([returnValue = 1](https://github.com/waydabber/m1ddc/blob/175cdf191df15ff4cff1cebef8c07f1d9f49bbf8/m1ddc.m#L72) isn't overwritten when [!avService](https://github.com/waydabber/m1ddc/blob/175cdf191df15ff4cff1cebef8c07f1d9f49bbf8/m1ddc.m#L203-L208)), but returns status code 0 when luminance is successfully retrieved.

Reply:
This should work then
```suggestion
    if ENV["CI"]
      assert_match "Could not find a suitable external display.", shell_output("#{bin}/m1ddc get luminance", 1)
    else
      assert_match /\d*/, shell_output("#{bin}/m1ddc get luminance")
    end    
```

Reply:
I'm not sure :( There seems to be some variance between the CI runners- https://github.com/Homebrew/homebrew-core/commit/9c1b7dda8a2117907149ab314a48eb3d98ea86b0 tried just using `shell_output("#{bin}/m1ddc get luminance")` and it passed on [tests (12-arm64)](https://github.com/Homebrew/homebrew-core/actions/runs/3868771519/jobs/6594473285#logs) but failed for [tests (13-arm64-3868771519-1)](https://github.com/Homebrew/homebrew-core/actions/runs/3868771519/jobs/6594473212).

I expect the opposite would unfortunately happen with this suggested change (failing on 12 but succeeding for 13)- the ideal here would be asserting slightly different stdout based on the return code.

Reply:
That's very weird, I would expect this variable to be set on all our CI.

CC @Bo98 who might know if it should be